pandas/core/groupby/groupby.py

```diff
@@ -2433,21 +2447,40 @@ def _get_cythonized_result(
             name = obj.name
             values = obj._values
 
+            if numeric_only and not is_numeric_dtype(values):
+                continue
+
             if aggregate:
                 result_sz = ngroups
             else:
                 result_sz = len(values)
 
-            result = np.zeros(result_sz, dtype=cython_dtype)
-            func = partial(base_func, result, labels)
+            if needs_2d:
```

Comment:
can you use at_least2d or just reshape here?

Reply:
can you do this here

Reply:
I think you're asking to replace

    if needs_2d:
        result = np.zeros((result_sz, 1), dtype=cython_dtype)
    else:
        result = np.zeros(result_sz, dtype=cython_dtype)

with

    result = np.zeros(result_sz, dtype=cython_dtype)
    if needs_2d:
        result = np.reshape((-1, 1), dtype=cython_dtype)

I think the np.reshape version is less performant when `needs_2d` is `True`, no?

Reply:
I don't believe at_least2d is applicable, it will turn a 1d into a single row (1xn) whereas we need a column (nx1).

Reply:
```
result = np.zeros(result_sz, dtype=cython_dtype)
if needs_2d:
    result = result.reshape((-1, 1))
```

yes this would be an improvement